### PR TITLE
Refactor Python local exceptions

### PR DESCRIPTION
**Pull request by pepone**
This PR simplifies the conversion of C++ exceptions to Python exceptions, similar to changes done by @bernardnormier to MATLAB, Ruby, and PHP

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]
Looks good!

File: python/python/Ice/LocalExceptions.py [hunk 1]
```diff
@@ -3,108 +3,61 @@
...
     """
 
-    def __init__(self, reason=""):
-        self.reason = reason
-
-    def __str__(self):
-        return "::Ice::InitializationException"
+    def __init__(self, id=None, facet= "", operation="", msg=""):
+        LocalException.__init__(self, msg)
+        self.id = id
```
[the remark this answers is not in the document]
Comment (pepone) [reply python/python/Ice/LocalExceptions.py]: fixed

File: python/python/Ice/LocalExceptions.py [hunk 2]
```diff
@@ -135,544 +81,280 @@ class UnknownUserException(UnknownException):
...
-
-    def __str__(self):
-        return "::Ice::ConnectFailedException"
-
     __module__ = "Ice"
     __class__ = "ConnectFailedException"
 
 
-class ConnectionRefusedException(ConnectFailedException):
+class ConnectionLostException(SocketException):
```
[the remark this answers is not in the document]
Comment (pepone) [reply python/python/Ice/LocalExceptions.py:184]: Removed, also replaced `__module="Ice"` in each class by `__name__ = "Ice"` at the beginning of the file.

File: python/python/Ice/LocalExceptions.py [hunk 3]
```diff
@@ -135,544 +81,280 @@ class UnknownUserException(UnknownException):
...
-
-    def __str__(self):
-        return "::Ice::CommunicatorDestroyedException"
-
     __module__ = "Ice"
-    __class__ = "CommunicatorDestroyedException"
 
 
-class ObjectAdapterDeactivatedException(LocalException):
+class ProtocolException(LocalException):
```
[the remark this answers is not in the document]
Comment (pepone) [reply python/python/Ice/LocalExceptions.py:98]: Added comments

File: python/python/Ice/LocalExceptions.py [hunk 4]
```diff
@@ -3,108 +3,61 @@
...
+        self.id = id
+        self.facet = facet
+        self.operation = operation
 
     __module__ = "Ice"
-    __class__ = "InitializationException"
-
 
-class AlreadyRegisteredException(LocalException):
+class ObjectNotExistException(RequestFailedException):
```
[the remark this answers is not in the document]
Comment (pepone) [reply python/python/Ice/LocalExceptions.py:47]: added @final

**Review by externl — APPROVED** [review]

File: python/test/Ice/properties/Client.py [hunk 5]
```diff
@@ -106,7 +106,7 @@ def run(sef, args):
             properties = Ice.createProperties()
             properties.getIceProperty("Ice.UnknownProperty")
             test(False)
-        except Ice.UnknownException:
+        except Ice.LocalException:
```
[the remark this answers is not in the document]
Comment (pepone) [reply python/test/Ice/properties/Client.py:119]: Added

**Pull request merged**